### ISSUE TITLE
Application goes down on startup when it can not connect to rabbitmq

### ISSUE THREAD

**Issue by hannaf** [conduitframework/conduit_amqp#7]
I'm trying to configure conduit in a project. It works fine for most cases, but one of the problems I'm having is that once the connection with rabbitmq can't be established for some reason, the application goes down. The issue doesn't happen when the connection is lost after the application is already up.

The error occurs at `ConduitAMQP.PubSub` module, line 23, when it tries to get a connection from the pool but receives an error. I'm not sure, but this step seems needed only if the 'cofigure' macro has been used. One possible solution to avoid this problem, at least on this case, would be to verify if this step must be executed by changing the code as above:

Before:
```elixir
    case ConduitAMQP.with_conn(broker, &Channel.open/1) do
      {:ok, chan} -> configure(chan, topology)
    end
```

After:
```elixir
    unless Enum.empty?(topology) do 
      {:ok, chan} = ConduitAMQP.with_conn(broker, &Channel.open/1)
      configure(chan, topology)
    end
```

This solution solves my problem, but I'm not sure it's the best solution, if the configure macro it's set up the problem will persist and I don't know how would be the best way to deal with this.

**Comment by blatyo** [comment]
You're correct, ConduitAMQP expects there to be a connection at startup so that it can configure queues and exchanges. It's not ideal. The way this is solved in ConduitSQS, is that it starts a [transient process](https://github.com/conduitframework/conduit_sqs/blob/master/lib/conduit_sqs/setup.ex#L19) that [runs the setup](https://github.com/conduitframework/conduit_sqs/blob/master/lib/conduit_sqs/setup.ex#L42) and then [sets a value in an ETS table](https://github.com/conduitframework/conduit_sqs/blob/master/lib/conduit_sqs/setup.ex#L43) that [the consumers check for](https://github.com/conduitframework/conduit_sqs/blob/master/lib/conduit_sqs/poller.ex#L88) before trying to consume from the queue. Something similar would ideally be done in ConduitAMQP. 

If you would like to take a shot at implementing that I can provide guidance.

**Comment by hannaf** [comment]
Yes, I would like to take a shot.

**Comment by blatyo** [comment]
Cool, let me know if you have any questions.

**Comment by blatyo** [comment]
@hannaf I'm going to start working on this.
